### PR TITLE
Fix RDDLSimAgent to correctly return the immediate reward to PROST

### PR DESCRIPTION
**Pull request by GMMDMDIDEMS**
Currently, `0.0` is always sent to PROST as immediate_reward (`msg = self.build_state_msg(state, turn, 0.0)`). This causes PROST to generate incorrect log files. Changing to `msg = self.build_state_msg(state, turn, reward)` fixes the issue.

I have also removed the `else` statement that was made unnecessary by `break` [Line 154] .

### CODE REVIEW

File: pyRDDLGym/Core/Policies/RDDLSimAgent.py [hunk 1]
```diff
@@ -127,32 +137,39 @@ def run_round(self, connection):
...
+                self.logs[-1].append(
+                    {
+                        "state": json.loads(
+                            str(state)
+                            .replace("'", '"')
+                            .replace("True", "true")
+                            .replace("False", "false")
+                        ),
+                        "actions": False,
+                        "round_reward": float(round_reward),
```
Comment (GMMDMDIDEMS) [inline pyRDDLGym/Core/Policies/RDDLSimAgent.py:149]: I have changed the order and rewritten `reward` to `round_reward`.

**Review by GMMDMDIDEMS — COMMENTED** [review]
Added some additional information regarding my changes.

**Comment by mike-gimelfarb** [comment]
Thanks for your pull request. Please note, that we are preparing some major changes in preparation for pyRDDLGym 2.0 on Feb 6, which will have a much cleaner and more accessible backend and frontend API for the user (as well as reorganization of the whole package), a vast collection of bug squashed, etc. Unfortunately it will not be backward compatible with 1.4. I'll review your changes and resolve conflicts with the new branch closer to Feb 6, since it will not be compatible with that branch as is, but your change as proposed looks to be critical to the RDDLSimAgent.

You can find the branch [here](https://github.com/ataitler/pyRDDLGym/tree/pyRDDLGym-v2-branch), in which case please feel free to suggest any fixes you think should also be incorporated into that branch before it is merged. We are always looking for testers to help ensure these changes are actually making things better :)

**Comment by GMMDMDIDEMS** [comment]
Thank you very much for the quick response. I have had a look at the branch. Is it intended behaviour that the `RDDL2Graph` class can only handle domains and instances that are defined within the pyRDDLGym package?

Are there other ways to find out the dbn structure (of domains/instances not defined within pyRDDLGym)?

**Comment by mike-gimelfarb** [comment]
For your first question, I replaced the (now obsolete) ExampleManager with the make() function only temporarily as a placeholder, and for running detailed tests to make sure the changes were successful, while we also sort out the remaining issues with the xadd subpackage (there are currently still a few), then I will generalize it. We are also making changes to rddlrepository in the next couple of days, so hopefully it will be compatible with the addition of new (user) domains. This is all in the works, stay tuned...

For your issue running it, yes when I tried to debug the api changes I merged today, I also ran into the same issue. It's strange given that it was running last year on an older environment. I suspect it is likely an internal conflict in the xadd package with latest sympy. I notified Jihwan (the original author of the xaddpy code) about the error in xaddpy, and he will help with fixing core.xadd either tomorrow or next week onwards.

**Comment by mike-gimelfarb** [comment]
I have merged your pull request manually into the v2 branch, and Ayal will later merge your PR into 1.4 before we freeze that version as reference.

**Review by ataitler — APPROVED** [review]

**Pull request merged**